sdks/python/apache_beam/utils/annotations.py

```diff
@@ -88,6 +88,14 @@ def exp_multiply(arg1, arg2):
 from functools import wraps
 
 
+class BeamDeprecationWarning(DeprecationWarning):
+  """Beam-specific deprecation warnings."""
+
+
+# Don't ignore BeamDeprecationWarnings.
+warnings.simplefilter('once', BeamDeprecationWarning)
```

Comment:
What does `'once'` do? If there are 2 warnings of the same type does it only show one of them?

Reply:
It shows only first occurrence of a matching warning. So if a function is decorated as deprecated, and called twice, we produce warning only once. If another function is also decorated as deprecated, we will still produce the first warning.
We used to have a test that verified this behavior albeit for experimental warnings: https://github.com/apache/beam/pull/6687/files#diff-ad168c748517e6a41781bd795008b4b5L127. The test didn't work on Python 3 because test framework on Python 3 calls `warnings.simplefilter('default', BeamDeprecationWarning)` at some point. We can revive the test and add `warnings.simplefilter('once', BeamDeprecationWarning)` to the test code make it work on Py3.

Reply:
Great thanks. This sounds like a good expected behavior.